subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -5369,6 +5374,43 @@ static int32_t lwm2m_engine_service(const int64_t timestamp)
 					      timestamp);
 }
 
+#if defined(CONFIG_LWM2M_QUEUE_MODE_ENABLED)
+
+int lwm2m_engine_close_socket_connection(struct lwm2m_ctx *client_ctx)
+{
+	int ret;
+
+	if (client_ctx->sock_fd >= 0) {
+		ret = close(client_ctx->sock_fd);
+		if (ret) {
+			LOG_ERR("Failed to close socket: %d", errno);
+			ret = -errno;
+			return ret;
+		}
+		client_ctx->sock_fd = -1;
+		client_ctx->connected = false;
+	}
+
+#ifdef CONFIG_LWM2M_DTLS_SUPPORT
+	if (client_ctx->use_dtls) {
+		client_ctx->sock_fd =
+			socket(client_ctx->remote_addr.sa_family, SOCK_DGRAM, IPPROTO_DTLS_1_2);
```

Comment:
Anyway, why is the socket reopened rightaway? It's a bit odd to me, that we're left with a misconfigured socket. Maybe, instead of having a separate `connected` flag it'd be enough to monitor `client_ctx->sock_fd`, and reopen/reconnect if it's set to -1?